File: libs/HandySales.Domain/Entities/RutaCarga.cs
using System.ComponentModel.DataAnnotations.Schema;
using HandySales.Domain.Common;

namespace HandySales.Domain.Entities;

[Table("RutasCarga")]
public class RutaCarga : AuditableEntity
{
    [Column("id")]
    public int Id { get; set; }

    [Column("ruta_id")]
    public int RutaId { get; set; }

    [Column("producto_id")]
    public int ProductoId { get; set; }

    [Column("tenant_id")]
    public int TenantId { get; set; }

    [Column("cantidad_entrega")]
    public int CantidadEntrega { get; set; }

    [Column("cantidad_venta")]
    public int CantidadVenta { get; set; }

    [Column("cantidad_total")]
    public int CantidadTotal { get; set; }

    [Column("precio_unitario")]
    public double PrecioUnitario { get; set; }

    // Navigation properties
    public RutaVendedor Ruta { get; set; } = null!;
    public Producto Producto { get; set; } = null!;
    public Tenant Tenant { get; set; } = null!;
}

File: apps/mobile-app/src/db/models/RutaCarga.ts
import { Model } from '@nozbe/watermelondb';
import { field, date, readonly, relation } from '@nozbe/watermelondb/decorators';

/**
 * Producto suelto cargado en el camión para venta directa (junction RutasCarga
 * en backend). Read-only en mobile — admin define la carga desde web. Mobile
 * solo lo muestra para que el vendedor sepa qué inventario lleva disponible.
 *
 * Cantidades:
 * - cantidadEntrega: para entregar pedidos asignados (RutasPedidos)
 * - cantidadVenta: para venta directa libre durante la ruta
 * - cantidadTotal: suma de ambas (lo que físicamente sale al camión)
 */
export default class RutaCarga extends Model {
  static table = 'ruta_carga';

  static associations = {
    rutas: { type: 'belongs_to' as const, key: 'ruta_id' },
    productos: { type: 'belongs_to' as const, key: 'producto_id' },
  };

  @field('server_id') serverId!: number;
  @field('ruta_id') rutaId!: string;
  @field('producto_id') productoId!: string;
  @field('producto_server_id') productoServerId!: number;
  @field('cantidad_entrega') cantidadEntrega!: number;
  @field('cantidad_venta') cantidadVenta!: number;
  @field('cantidad_total') cantidadTotal!: number;
  @field('precio_unitario') precioUnitario!: number;
  @field('activo') activo!: boolean;
  @readonly @date('created_at') createdAt!: Date;
  @date('updated_at') updatedAt!: Date;

  @relation('rutas', 'ruta_id') ruta: any;
  @relation('productos', 'producto_id') producto: any;
}
